### ISSUE TITLE
promclient: TestConfiguredFlags_e2e test failing 

### ISSUE THREAD

**Issue by Chandranana** [thanos-io/thanos#5752]
**Thanos, Prometheus and Golang version used**:
Thanos v0.11.0
Prometheus v2.37.1
Go v1.19

**Output of "thanos --version" or docker image:tag used.**
thanos --version
thanos, version  (branch: , revision: )
  build user:
  build date:
  go version:       go1.19

**Object Storage Provider**:

**What happened**:
TestConfiguredFlags_e2e test failing 

**What you expected to happen**:
Test should pass.

**How to reproduce it (as minimally and precisely as possible)**:
Execute make test command 

**Full logs to relevant components**:

</details>
</p>

```
promclient_e2e_test.go:70: unexpected error: request config against http://localhost:44748/api/v1/status/flags: Get "http://localhost:44748/api/v1/status/flags": EOF

--- FAIL: TestConfiguredFlags_e2e (2.03s)
    --- FAIL: TestConfiguredFlags_e2e/v2.4.3 (2.03s)
FAIL
FAIL    github.com/thanos-io/thanos/pkg/promclient      45.056s
```

</p>
</details>

**Environment**:
- OS (e.g. from /etc/os-release): rhel 8.5

**Please let me know if any dependency needs to be installed to resolve this issue or is there any prerequisite setup?**

**Comment by Chandranana** [comment]
Hi Team, Please share if there are any pointers on this issue. Please let us know if any information is needed from our side.

**Comment by matej-g** [comment]
Hey @Chandranana I'm not sure what exactly is the problem here? Test seems to work fine on my local. Are you really on Thanos v0.11.0?

**Comment by Chandranana** [comment]
Yes @matej-g I have installed v0.11.0. I am running the tests on power architecture. The master branch test execution works fine but not this version.

**Comment by Chandranana** [comment]
@matej-g Also i am facing issue installing the k8s.io.klog dependency required for running tests. I was able to install this dependency a few weeks back.
I have raised the issue for the same - https://github.com/thanos-io/thanos/issues/5765. Please let me know if this has changed.

**Comment by Chandranana** [comment]
@matej-g Any update on this issue, please let me know.
